libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/KotlinNativeTargetPreset.kt

```diff
@@ -88,12 +89,14 @@ abstract class AbstractKotlinNativeTargetPreset<T : KotlinNativeTarget>(
 
         createTargetConfigurator().configureTarget(result)
 
-        SingleActionPerProject.run(project, "setUpKotlinNativePlatformDependencies") {
-            project.gradle.addListener(object : BuildAdapter() {
-                override fun projectsEvaluated(gradle: Gradle) {
-                    project.setUpKotlinNativePlatformDependencies()
-                }
-            })
+        if (!isConfigurationCacheAvailable(project.gradle)) {
+            SingleActionPerProject.run(project, "setUpKotlinNativePlatformDependencies") {
+                project.gradle.addListener(object : BuildAdapter() {
```

Comment:
You can't use `gradle.addListener(BuildListener)` but in that case you only need to react to `projectsEvaluated` and that can be done using `gradle.projectsEvaluated(action)`.